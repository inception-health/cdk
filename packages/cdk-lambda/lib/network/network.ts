import * as cdk from "aws-cdk-lib";
import {
  Vpc,
  SubnetType,
  SecurityGroup,
  Peer,
  Port,
  IVpc,
  ISecurityGroup,
  FlowLog,
  FlowLogTrafficType,
  FlowLogDestination,
} from "aws-cdk-lib/aws-ec2";
import { ILogGroup } from "aws-cdk-lib/aws-logs";
import { Construct } from "constructs";

import { Lambda } from "../lambda";

export interface StandardNetworkProps {
  encryption: cdk.aws_kms.IKey;
  maxAzs?: number;
  name?: string;
}

export interface StandardNetworkCdk {
  flowLogGroup: ILogGroup;
  flowLogRole: cdk.aws_iam.IRole;
  flowLogs: FlowLog;
  lambdaSecurityGroup: ISecurityGroup;
  vpc: IVpc;
}

const defaultProps: Partial<StandardNetworkProps> = {
  // Ensure high availability by spreading across 3 availability zones
  maxAzs: 3,
};

export class StandardNetwork extends Construct {
  public readonly cdk: StandardNetworkCdk;

  private readonly props: StandardNetworkProps;

  constructor(scope: Construct, id: string, props?: StandardNetworkProps) {
    super(scope, id);

    // Initialize the properties
    this.cdk = {} as StandardNetworkCdk;
    this.props = Object.assign({}, defaultProps, props);

    // Create the VPC
    this.cdk.vpc = new Vpc(this, "vpc", {
      maxAzs: this.props.maxAzs,
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: "PublicSubnet",
          subnetType: SubnetType.PUBLIC,
          // (Control IDs: 164.308(a)(3)(i), 164.308(a)(4)(ii)(A),
          // 164.308(a)(4)(ii)(C), 164.312(a)(1), 164.312(e)(1)). Manage access
          // to the AWS Cloud by ensuring Amazon Virtual Private Cloud (VPC)
          // subnets are not automatically assigned a public IP address.
          mapPublicIpOnLaunch: false,
        },
        {
          cidrMask: 24,
          name: "PrivateSubnet",
          subnetType: SubnetType.PRIVATE_WITH_EGRESS,
        },
        {
          cidrMask: 28,
          name: "IsolatedSubnet",
          subnetType: SubnetType.PRIVATE_ISOLATED,
        },
      ],
    });

    // Create a security group for the Lambda function
    this.cdk.lambdaSecurityGroup = new SecurityGroup(
      this,
      "LambdaSecurityGroup",
      {
        vpc: this.cdk.vpc,
        description: "Security group for Lambda function",
        allowAllOutbound: true,
      },
    );

    // Restrict inbound traffic to the security group
    this.cdk.lambdaSecurityGroup.addIngressRule(
      Peer.anyIpv4(),
      Port.tcp(443),
      "Allow HTTPS traffic",
    );
    this.cdk.lambdaSecurityGroup.addIngressRule(
      Peer.anyIpv4(),
      Port.tcp(80),
      "Allow HTTP traffic",
    );

    // Create a log group for the VPC flow logs
    this.cdk.flowLogGroup = new cdk.aws_logs.LogGroup(this, "FlowLogGroup", {
      logGroupName: `/aws/vpc/flow-logs/${this.cdk.vpc.vpcId}`,
      retention: cdk.aws_logs.RetentionDays.ONE_MONTH,
      encryptionKey: this.props.encryption,
    });

    // Create a flow log for the VPC
    this.cdk.flowLogs = this.cdk.vpc.addFlowLog("FlowLog", {
      trafficType: FlowLogTrafficType.ALL,
      destination: FlowLogDestination.toCloudWatchLogs(this.cdk.flowLogGroup),
    });

    // Output the VPC ID and security group ID
    new cdk.CfnOutput(this, "VpcId", {
      value: this.cdk.vpc.vpcId,
    });

    new cdk.CfnOutput(this, "LambdaSecurityGroupId", {
      value: this.cdk.lambdaSecurityGroup.securityGroupId,
    });
  }

  public get vpc(): IVpc {
    return this.cdk.vpc;
  }

  public get lambdaSecurityGroup(): ISecurityGroup {
    return this.cdk.lambdaSecurityGroup;
  }

  public attachLambda(lambda: Lambda) {
    // lambda.cdk.function.connections.allowTo(
    //   this.cdk.lambdaSecurityGroup,
    //   Port.allTraffic(),
    //   "Allow Lambda to access resources in the VPC",
    // );
  }
}
