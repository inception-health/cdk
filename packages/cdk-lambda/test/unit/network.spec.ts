import * as cdk from "aws-cdk-lib";
import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";

import { StandardNetwork } from "../../lib/network/network";

describe("StandardVpc", () => {
  let app: App;
  let stack: cdk.Stack;
  let template: Template;
  let vpc: StandardNetwork;
  let key: cdk.aws_kms.IKey;

  beforeEach(() => {
    app = new App();
    stack = new cdk.Stack(app, "TestStack");
    key = new cdk.aws_kms.Key(stack, "Key");
    vpc = new StandardNetwork(stack, "TestVpc", {
      name: "TestVpc",
      encryption: key,
    });
    template = Template.fromStack(stack);
  });

  test("Matches the Snapshot", () => {
    // THEN
    expect(template).toMatchSnapshot();
  });

  test("VPC is created with correct properties", () => {
    template.hasResourceProperties("AWS::EC2::VPC", {
      CidrBlock: "10.0.0.0/16",
      EnableDnsSupport: true,
      EnableDnsHostnames: true,
    });
  });

  test("VPC has 6 subnets with correct configuration", () => {
    template.resourceCountIs("AWS::EC2::Subnet", 6);
    // Public subnets
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.0.0/24",
      MapPublicIpOnLaunch: false,
    });
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.1.0/24",
      MapPublicIpOnLaunch: false,
    });

    // Private subnets
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.2.0/24",
      MapPublicIpOnLaunch: false,
    });
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.3.0/24",
      MapPublicIpOnLaunch: false,
    });

    // Isolated subnets
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.4.0/28",
      MapPublicIpOnLaunch: false,
    });
    template.hasResourceProperties("AWS::EC2::Subnet", {
      CidrBlock: "10.0.4.16/28",
      MapPublicIpOnLaunch: false,
    });
  });

  test("Security group for Lambda is created with correct properties", () => {
    const vpcIdRef = stack.getLogicalId(
      vpc.cdk.vpc.node.defaultChild as cdk.CfnElement,
    );

    template.hasResourceProperties("AWS::EC2::SecurityGroup", {
      GroupDescription: "Security group for Lambda function",
      VpcId: {
        Ref: vpcIdRef,
      },
    });
  });

  test("Security group allows inbound HTTP and HTTPS traffic", () => {
    template.hasResourceProperties("AWS::EC2::SecurityGroup", {
      SecurityGroupIngress: [
        {
          CidrIp: "0.0.0.0/0",
          Description: "Allow HTTPS traffic",
          FromPort: 443,
          IpProtocol: "tcp",
          ToPort: 443,
        },
        {
          CidrIp: "0.0.0.0/0",
          Description: "Allow HTTP traffic",
          FromPort: 80,
          IpProtocol: "tcp",
          ToPort: 80,
        },
      ],
    });
  });

  test("Flow log group is created with correct properties", () => {
    const vpcIdRef = stack.getLogicalId(
      vpc.cdk.vpc.node.defaultChild as cdk.CfnElement,
    );

    template.hasResourceProperties("AWS::Logs::LogGroup", {
      LogGroupName: {
        "Fn::Join": [
          "",
          [
            "/aws/vpc/flow-logs/",
            {
              Ref: vpcIdRef,
            },
          ],
        ],
      },
      RetentionInDays: 30,
    });
  });
});
