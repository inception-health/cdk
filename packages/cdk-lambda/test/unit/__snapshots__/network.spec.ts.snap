// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardVpc Matches the Snapshot 1`] = `
{
  "Outputs": {
    "TestVpcLambdaSecurityGroupId222B1947": {
      "Value": {
        "Fn::GetAtt": [
          "TestVpcLambdaSecurityGroup3CD5C7B9",
          "GroupId",
        ],
      },
    },
    "TestVpcVpcIdA46ACB31": {
      "Value": {
        "Ref": "TestVpcvpc290DF0A0",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Key961B73FD": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "TestVpcFlowLogGroupB8F285E7": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "Key961B73FD",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/vpc/flow-logs/",
              {
                "Ref": "TestVpcvpc290DF0A0",
              },
            ],
          ],
        },
        "RetentionInDays": 30,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "TestVpcLambdaSecurityGroup3CD5C7B9": {
      "Properties": {
        "GroupDescription": "Security group for Lambda function",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow HTTP traffic",
            "FromPort": 80,
            "IpProtocol": "tcp",
            "ToPort": 80,
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "TestVpcvpc290DF0A0": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "TestVpcvpcFlowLog092DA252": {
      "Properties": {
        "DeliverLogsPermissionArn": {
          "Fn::GetAtt": [
            "TestVpcvpcFlowLogIAMRole847A5B05",
            "Arn",
          ],
        },
        "LogDestinationType": "cloud-watch-logs",
        "LogGroupName": {
          "Ref": "TestVpcFlowLogGroupB8F285E7",
        },
        "ResourceId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
        "ResourceType": "VPC",
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/FlowLog",
          },
        ],
        "TrafficType": "ALL",
      },
      "Type": "AWS::EC2::FlowLog",
    },
    "TestVpcvpcFlowLogIAMRole847A5B05": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "vpc-flow-logs.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/FlowLog",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestVpcvpcFlowLogIAMRoleDefaultPolicy25209C07": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
                "logs:DescribeLogStreams",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "TestVpcFlowLogGroupB8F285E7",
                  "Arn",
                ],
              },
            },
            {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "TestVpcvpcFlowLogIAMRole847A5B05",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TestVpcvpcFlowLogIAMRoleDefaultPolicy25209C07",
        "Roles": [
          {
            "Ref": "TestVpcvpcFlowLogIAMRole847A5B05",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TestVpcvpcIGW907A36FB": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "TestVpcvpcIsolatedSubnetSubnet1RouteTable4051727A": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/IsolatedSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcIsolatedSubnetSubnet1RouteTableAssociation64D8CAE0": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcIsolatedSubnetSubnet1RouteTable4051727A",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcIsolatedSubnetSubnet1Subnet22105435",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcIsolatedSubnetSubnet1Subnet22105435": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.4.0/28",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "IsolatedSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/IsolatedSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcIsolatedSubnetSubnet2RouteTable7AB4D616": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/IsolatedSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcIsolatedSubnetSubnet2RouteTableAssociation6D15E2A0": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcIsolatedSubnetSubnet2RouteTable7AB4D616",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcIsolatedSubnetSubnet2Subnet187B2F5A",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcIsolatedSubnetSubnet2Subnet187B2F5A": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.4.16/28",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "IsolatedSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/IsolatedSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcPrivateSubnetSubnet1DefaultRoute7060A050": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet1NATGateway35B38E48",
        },
        "RouteTableId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet1RouteTableCC3231A8",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TestVpcvpcPrivateSubnetSubnet1RouteTableAssociation7DD8C881": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet1RouteTableCC3231A8",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet1SubnetED876DDD",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcPrivateSubnetSubnet1RouteTableCC3231A8": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PrivateSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcPrivateSubnetSubnet1SubnetED876DDD": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.2.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PrivateSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PrivateSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcPrivateSubnetSubnet2DefaultRoute689641E6": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet2NATGateway195E3CF3",
        },
        "RouteTableId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet2RouteTable954F4431",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TestVpcvpcPrivateSubnetSubnet2RouteTable954F4431": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PrivateSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcPrivateSubnetSubnet2RouteTableAssociationD470E5F3": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet2RouteTable954F4431",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPrivateSubnetSubnet2Subnet5790C11E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcPrivateSubnetSubnet2Subnet5790C11E": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.3.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PrivateSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PrivateSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcPublicSubnetSubnet1DefaultRouteF3CACBFC": {
      "DependsOn": [
        "TestVpcvpcVPCGW555F9039",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "TestVpcvpcIGW907A36FB",
        },
        "RouteTableId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet1RouteTable06E8ADBF",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TestVpcvpcPublicSubnetSubnet1EIPCD0FFFD6": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "TestVpcvpcPublicSubnetSubnet1NATGateway35B38E48": {
      "DependsOn": [
        "TestVpcvpcPublicSubnetSubnet1DefaultRouteF3CACBFC",
        "TestVpcvpcPublicSubnetSubnet1RouteTableAssociation32F6ABB5",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "TestVpcvpcPublicSubnetSubnet1EIPCD0FFFD6",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet1Subnet076622E8",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "TestVpcvpcPublicSubnetSubnet1RouteTable06E8ADBF": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcPublicSubnetSubnet1RouteTableAssociation32F6ABB5": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet1RouteTable06E8ADBF",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet1Subnet076622E8",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcPublicSubnetSubnet1Subnet076622E8": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PublicSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcPublicSubnetSubnet2DefaultRoute89C66B87": {
      "DependsOn": [
        "TestVpcvpcVPCGW555F9039",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "TestVpcvpcIGW907A36FB",
        },
        "RouteTableId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet2RouteTable282D1ED6",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TestVpcvpcPublicSubnetSubnet2EIP75A7EFE4": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "TestVpcvpcPublicSubnetSubnet2NATGateway195E3CF3": {
      "DependsOn": [
        "TestVpcvpcPublicSubnetSubnet2DefaultRoute89C66B87",
        "TestVpcvpcPublicSubnetSubnet2RouteTableAssociationEF60F380",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "TestVpcvpcPublicSubnetSubnet2EIP75A7EFE4",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet2SubnetF7EFD1B9",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "TestVpcvpcPublicSubnetSubnet2RouteTable282D1ED6": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TestVpcvpcPublicSubnetSubnet2RouteTableAssociationEF60F380": {
      "Properties": {
        "RouteTableId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet2RouteTable282D1ED6",
        },
        "SubnetId": {
          "Ref": "TestVpcvpcPublicSubnetSubnet2SubnetF7EFD1B9",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TestVpcvpcPublicSubnetSubnet2SubnetF7EFD1B9": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.1.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PublicSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "TestStack/TestVpc/vpc/PublicSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TestVpcvpcVPCGW555F9039": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "TestVpcvpcIGW907A36FB",
        },
        "VpcId": {
          "Ref": "TestVpcvpc290DF0A0",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
